pkg/api/v1beta1/types.go

```diff
@@ -1625,3 +1623,109 @@ type ComponentStatusList struct {
 
 	Items []ComponentStatus `json:"items" description:"list of component status objects"`
 }
+
+// SecurityContext holds security configuration that will be applied to a container.  If a security context
+// is set on the container spec then it must comply with any constraints defined in the SecurityConstraints context
+// it is running in.  If a security context is not supplied and the pod is running under a SecurityConstraints context
+// then a default SecurityContext may be applied.
+type SecurityContext struct {
+	// Capabilities are the capabilities to add/drop when running the container
+	// TODO: will need to refactor this from the container spec to here
+	Capabilities *Capabilities `json:"capabilities,omitempty"`
+
+	// Run the container in privileged mode
+	// TODO: will need to refactor this from the container spec to here
+	Privileged bool `json:"privileged,omitempty"`
+
+	// SELinuxOptions are the labels to be applied to the container
+	// and volumes
+	SELinuxOptions *SELinuxOptions `json:"seLinuxOptions,omitempty"`
+
+	// RunAsUser is the UID to run the entrypoint of the container process.  Corresponding option is --user or -u
```

Comment:
Yeah.  Do we want to wait to add it until there is an implementation that can provide it?

----- Original Message -----
> > +// it is running in.  If a security context is not supplied and the pod is
> > running under a SecurityConstraints context
> > +// then a default SecurityContext may be applied.
> > +type SecurityContext struct {
> > +	// Capabilities are the capabilities to add/drop when running the
> > container
> > +	// TODO: will need to refactor this from the container spec to here
> > +	Capabilities *Capabilities `json:"capabilities,omitempty"`
> > +
> > +	// Run the container in privileged mode
> > +	// TODO: will need to refactor this from the container spec to here
> > +	Privileged bool `json:"privileged,omitempty"`
> > +
> > +	// SELinuxOptions are the labels to be applied to the container
> > +	// and volumes
> > +	SELinuxOptions *SELinuxOptions `json:"seLinuxOptions,omitempty"`
> > +
> > +	// RunAsUser is the UID to run the entrypoint of the container process.
> > Corresponding option is --user or -u
> 
> Corresponding option in what tool?  Docker?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/6287/files#r28996781